src/keyshot_command_handler.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+"""
+This script runs inside KeyShot. It accepts JSON commands from STDIN and executes them.
+"""
+
+import json
+import sys
+import traceback
+import lux  # noqa: F401
+
+
+def apply_render_device_override(render_device, override_render_device):
+    current_engine = lux.getRenderEngine()
+    is_gpu = current_engine in [lux.RENDER_ENGINE_PRODUCT_GPU, lux.RENDER_ENGINE_INTERIOR_GPU]
+    current_device = "GPU" if is_gpu else "CPU"
+    original_render_device_param = render_device
+
+    engine_names = {
+        0: "RENDER_ENGINE_PRODUCT",
+        1: "RENDER_ENGINE_INTERIOR",
+        3: "RENDER_ENGINE_PRODUCT_GPU",
+        4: "RENDER_ENGINE_INTERIOR_GPU",
+    }
+
+    # Apply render device override logic
+    if override_render_device:
+        print("Option to override the Keyshot render device was selected.")
+        if current_device == render_device:
+            print(
+                f"KeyShot render device {render_device} is already selected. Chosen override option matches Keyshot setting."
+            )
+        else:
+            print(f"Overriding {current_device} with {render_device}...")
+
+        # Apply render device changes
+        if render_device == "GPU" and current_device == "CPU":
+            engine_map = {
+                lux.RENDER_ENGINE_PRODUCT: lux.RENDER_ENGINE_PRODUCT_GPU,
+                lux.RENDER_ENGINE_INTERIOR: lux.RENDER_ENGINE_INTERIOR_GPU,
+            }
+            if current_engine in engine_map:
+                try:
+                    lux.setRenderEngine(engine_map[current_engine])
+                except Exception:
+                    raise RuntimeError(
+                        "GPU rendering was requested but no compatible GPU is available on this worker. Please manually set min GPUs to 1 under 'Host requirements' in the KeyShot integrated submitter."
+                    )
+        elif render_device == "CPU" and current_device == "GPU":
+            engine_map = {
+                lux.RENDER_ENGINE_PRODUCT_GPU: lux.RENDER_ENGINE_PRODUCT,
+                lux.RENDER_ENGINE_INTERIOR_GPU: lux.RENDER_ENGINE_INTERIOR,
+            }
+            if current_engine in engine_map:
+                lux.setRenderEngine(engine_map[current_engine])
```

Comment:
I see that we do error handling on line 47, should this part need error handling as well? 👀

Reply:
Fixed by combining the two cases